### ISSUE TITLE
Completion does not autocomplete filenames

### ISSUE THREAD

**Issue by ahmetalpbalkan** [Azure/azure-xplat-cli#1955]
I just added `azure --completion` script to my `~/.zshrc` and tried to deploy a template using command. I typed:

    azure group deployment create -n "myName" -f 

then I hit `<tab>` `<tab>` and nothing happened. So I had to Ctrl+C and run a `ls` to see what the file was actually named. I think the completion should somehow detect `-f` (and likewise `-e` in this command) is a filename input and do a glob in the PWD.

**Comment by devigned** [comment]
That would be really helpful.

**Comment by yugangw-msft** [comment]
I will take a look.

**Comment by ahmetalpbalkan** [comment]
@yugangw-msft can you please take a look here? This does not fix the bug entirely

0.9.10:

    $ azure vm extension set [bla bla] --public-config-path <tab><tab> <-- does not work

can you please reopen?

**Comment by yugangw-msft** [comment]
@ahmetalpbalkan, my PR introduced a new file related flag so auto-complete can recognize as a file related option and do a "ls". At that time I did convert some existing options though, but obviously i could not know all of them. 
Re-think now, maybe i don't even need to introduce a file flag; rather i should just do a "ls", even for a not file related option. Not that experienced on non-windows platform, but 2 reasons:
1. In mac I tried out several well-known commands, looks like a built-in behavior that 2 tabs means "ls".
2. We will have no need to convert file related options in azure-cli, which might have lot.
